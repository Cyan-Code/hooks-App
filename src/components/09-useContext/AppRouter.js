import React from 'react'
import { NavBar } from './NavBar';
import {AboutScreen} from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';

import {
  BrowserRouter as Router,
  Switch,
  Route,
} from "react-router-dom";

export const AppRouter = () => {
  return (
    <Router>
      <div>

        <NavBar/>
        <div className="container">
          <Switch>
            <Route exact path="/about">
              <AboutScreen/>
            </Route>
            <Route exact path="/login">
              <LoginScreen/>
            </Route>
            <Route exact path = "/">
              <HomeScreen/>
            </Route>
            <Route component = {HomeScreen}/>
          </Switch>
        </div>
      </div>
    </Router>
  )
}
