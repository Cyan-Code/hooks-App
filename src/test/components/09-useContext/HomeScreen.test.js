import { shallow } from 'enzyme';
import React from 'react';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas sobre el componente <HomeScreen />', () => {
  
  const user = {
    name:'Luis',
    email: 'luis@luis.com'
  }

  const wrapper = shallow(
    <UserContext.Provider value = {{
      user
    }}>
      <HomeScreen />
    </UserContext.Provider>
  )
  test('Debe de mostrarse correctamente', () => {
    expect( wrapper ).toMatchSnapshot()
  })
  
})


