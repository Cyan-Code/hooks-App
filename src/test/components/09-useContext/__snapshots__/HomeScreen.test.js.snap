// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <HomeScreen /> Debe de mostrarse correctamente 1`] = `<HomeScreen />`;
